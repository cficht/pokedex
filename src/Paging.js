import React, { Component } from 'react';

export default class Paging extends Component {
    state = 
    { page: 1, 
      perPage: 20,
    }

    changePage(value) {
        this.setState({ page: this.state.page + value})
        const queryString = window.location.hash.slice(1);
        const pageParams = new URLSearchParams(queryString);
        pageParams.set('page', this.state.page + value);
        pageParams.set('perPage', this.state.perPage)
        const paramToSet = pageParams.toString();
        window.location.hash = paramToSet;
    }

    componentDidMount() {
        const pageParams = new URLSearchParams();
        pageParams.set('page', this.state.page);
        const paramToSet = pageParams.toString();
        window.location.hash = paramToSet;

        window.addEventListener('hashchange', () => {
            const queryString = window.location.hash.slice(1);
            const pageParams = new URLSearchParams(queryString);
            let pageToUse = this.state.page;
            const parsedPage = Number(pageParams.get('page'));
            
            if (isNaN(parsedPage)) {
              pageToUse = 1;
            } else {
              pageToUse = parsedPage;
            }       
            this.setState({ page: pageToUse });
        })
    }
    
    render() {
        return (
            <div>
                <button onClick={() => this.changePage(-1)} disabled={this.state.page === 1}> LAST </button>
                <button onClick={() => this.changePage(1)} disabled={this.state.page === this.props.totalPage}> NEXT </button>
                <h5>Page {this.state.page}/{this.props.totalPage}</h5>
            </div>
        )
    }
}