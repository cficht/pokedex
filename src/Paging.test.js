import React from 'react';
import Paging from './Paging';
import renderer from "react-test-renderer";

test('renders Paging.js correctly', () => {
  const tree = renderer
  .create(<Paging />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});