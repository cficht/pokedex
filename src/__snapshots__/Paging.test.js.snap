// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Paging.js correctly 1`] = `
<h5
  className="paging"
>
  No results! :(
</h5>
`;
